src/Aquifer.Jobs/Services/TranslationProcessors/FrenchTranslationProcessingService.cs

```diff
@@ -0,0 +1,148 @@
+
+        text = Regex.Replace(text, _nnbspAfterPattern, m => m.Value + NarrowNonBreakingSpace);
+        text = Regex.Replace(text, _nnbspBeforePattern, m => NarrowNonBreakingSpace + m.Value);
+
+        // The regex matches the value but not the character before/after, so if a space comes before/after we'll end up with two spaces.
+        text = text.Replace($"{NarrowNonBreakingSpace} ", NarrowNonBreakingSpace)
+            .Replace($" {NarrowNonBreakingSpace}", NarrowNonBreakingSpace)
+            // Replace this dash specifically with non-breaking dash
+            .Replace("J.-C.", "J.‑C.");
+
+        return Task.FromResult(text);
+    }
+
+    private static Task<string> SetCorrectBibleBookAbbreviation(string text)
+    {
+        text = BibleBookRegex()
+            .Replace(text,
+                m => m.Value.ToLower().Replace(NarrowNonBreakingSpace, " ") switch
+                {
+                    "gen" or "ge" => "Gn",
+                    "exod" => "Ex",
+                    "lev" or "lév" => "Lv",
+                    "nomb" => "Nb",
+                    "deut" => "Dt",
+                    "josh" => "Jos",
+                    "ju" or "juges" => "Jg",
+                    "ruth" => "Rt",
+                    "1 s" or "1 sa" or "1 sam" => "1S",
+                    "2 s" or "2 sa" or "2 sam" => "2S",
+                    "1 r" or "1 rois" => "1R",
+                    "2 r" or "2 rois" => "2R",
+                    "1 ch" or "1 chr" => "1Ch",
+                    "2 ch" or "2 chr" => "2Ch",
+                    "ezra" or "esdras" => "Esd",
+                    "neh" or "néh" => "Né",
+                    "job" => "Jb",
+                    "psa" or "pss" => "Ps",
+                    "prov" or "pro" or "prv" => "Pr",
+                    "ecc" => "Ec",
+                    "sg" => "Ct",
+                    "is" or "isa" or "és" => "Es",
+                    "jer" or "jér" => "Jr",
+                    "lam" => "Lm",
+                    "éz" => "Ez",
+                    "da" or "dan" => "Dn",
+                    "hos" => "Os",
+                    "joel" or "joël" or "joe" => "Jl",
+                    "amos" => "Am",
+                    "ob" => "Ab",
+                    "jonah" or "jonas" => "Jon",
+                    "mic" => "Mi",
+                    "nah" => "Na",
+                    "hab" => "Ha",
+                    "zeph" or "soph" => "So",
+                    "hag" => "Ag",
+                    "zech" or "zach" or "zec" or "zac" => "Za",
+                    "mal" => "Ml",
+                    "matt" => "Mt",
+                    "mk" or "marc" => "Mc",
+                    "tob" => "Tb",
+                    "1 macc" => "1\u202fM",
+                    "2 macc" => "2\u202fM",
+                    "sa" => "Sg",
+                    "sir" => "Si",
+                    "bar" => "Ba",
+                    "lk" or "luc" => "Lc",
+                    "jean" or "john" => "Jn",
```

Comment:
tl;dr: Concern: Bible book name detection is difficult. If we are detecting a string that is already an abbreviation then it seems pretty safe to do the conversion.  I'm thinking it would be best to remove the full words and full proper names and catch them with manual review.

Some of these might be too catch all, especially with the `ToLower()` comparison.
Examples:
* If any of these are an actual French word then they'll be replaced (I don't know if any of these apply).  This could potentially be solved by not doing a `ToLower()` assuming the abbreviations usually are capitalized but it still wouldn't help for usage of the word at the beginning of a sentence.
* In this case, `Jean` is a proper name which we don't always want to abbreviate (e.g. `St. Jean Baptiste` would become `St. Jn Baptiste` which isn't desired).  The same is potentially true for `Esdras` above.

Reply:
The regex accounts for this. There has to be a number followed by a space or period for it to match.

Reply:
![image](https://github.com/user-attachments/assets/f41828bc-07d9-4224-9ebe-2c9d1d164225)